test/testtokenlist.cpp

```diff
@@ -161,6 +162,15 @@ class TestTokenList : public TestFixture {
         TokenList tokenlist(&settingsDefault);
         tokenlist.createTokens(std::move(tokensP)); // do not assert
     }
+
+    void ast1() {
+        const std::string s = "('Release|x64' == 'Release|x64');";
+
+        TokenList tokenlist(&settings);
+        std::istringstream istr(s);
+        tokenlist.createTokens(istr, Standards::Language::C);
+        tokenlist.createAst(); // TODO: do not crash
```

Comment:
@chrchr-github Any idea why this might crash? It works when done in the context of `Tokenizer::simplifyTokens1()`. IMO the AST generation in `TokenList` should not depend on the simplifications.

Reply:
Looks like the parentheses are not linked (`Tokenizer::createLinks()`).

Reply:
The logic seems to exists in several others places:
`TokenList::insertTokens()`
`createTokenFromExpression()`
`clangimport::AstNode::addTypeTokens()`

I assume this is done to work around the issue I am currently encountering as the code in question is only using a `TokenList`.